node-zookeeper-client/index.d.ts

```diff
@@ -0,0 +1,169 @@
+// Type definitions for Jasmine 2.5
+// Project: https://github.com/alexguan/node-zookeeper-client
+// Definitions by: York Yao <https://github.com/plantain-00/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+/// <reference types="node" />
+
+import * as EventEmitter from "events";
+
+export class Id {
+    scheme: string;
+    id: string;
+    constructor(scheme: string, id: string);
+}
+
+export class ACL {
+    perms: number;
+    id: Id;
+    constructor(perms: number, id: Id);
+}
+
+export const Permission: {
+    READ: number,
+    WRITE: number,
+    CREATE: number,
+    DELETE: number,
+    ADMIN: number,
+    ALL: number,
+};
+
+export interface Stat {
+    czxid: number;
+    mzxid: number;
+    ctime: number;
+    mtime: number;
+    version: number;
+    cversion: number;
+    aversion: number;
+    ephemeralOwner: number;
+    dataLength: number;
+    numChildren: number;
+    pzxid: number;
+}
+
+export class State {
+    static DISCONNECTED: State;
+    static SYNC_CONNECTED: State;
+    static AUTH_FAILED: State;
+    static CONNECTED_READ_ONLY: State;
+    static SASL_AUTHENTICATED: State;
+    static EXPIRED: State;
+
+    name: string;
+    code: number;
+    constructor(name: string, code: number);
+    toString(): string;
+}
+
+export class Event {
+    static NODE_CREATED: number;
+    static NODE_DELETED: number;
+    static NODE_DATA_CHANGED: number;
+    static NODE_CHILDREN_CHANGED: number;
+    type: string;
+    name: string;
+    path: string;
+    constructor(type: string, name: string, path: string);
+    toString(): string;
+    getType(): string;
+    getName(): string;
+    getPath(): string;
+}
+
+export class Transaction {
```

Comment:
The top level `index.js` file doesn't actually export the constructor, so this should be an `interface`.